pandas/compat/bz2.py

```diff
@@ -0,0 +1,27 @@
+"""
+Patched ``BZ2File`` to handle pickle protocol 5.
+"""
+
+from __future__ import annotations
+
+import bz2
+import sys
+
+from pandas.compat._utils import flatten_buffer
+
+if sys.version_info < (3, 10):
```

Comment:
Could you use `not PY310:` (PY310 in `pandas/compat/__init__.py`)

Reply:
Am afraid this will lead to more cyclic `ImportError`s as noted above. Not seeing an easy way to resolve this

Reply:
Tried just copying over the `PY310` definition verbatim. Hopefully that makes `grep`ping easier even though it is now defined in two places. Maybe we could refactor out into a constants file that is more free of cyclic import issues.